### ISSUE TITLE
bug: ion-datetime with a min/max does not update valid values on first picker change

### ISSUE THREAD

**Issue by eisene** [ionic-team/ionic#12319]
**Ionic version:**  (check one with "x")
[ ] **1.x** (For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[x] **3.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
1. Place a DateTime component on a page with a minimum (or maximum) date in the middle of a month/year, for example 06/15/2013
2. In the model set the initial date that will be shown in the DateTime to the minimum date
3. Open the date picker and make exactly one change that makes the valid values in the picker columns to change. For example, change the year to 2014
4. The picker columns don't update, many valid dates are missing
5. Make one more change in the same column. For example, change the year to 2015
6. Everything updates and the picker is now in a correct state

**Expected behavior:**
The valid days and months should update at step 4

**Steps to reproduce:**
http://plnkr.co/edit/eHapiVZtvMqjjsTJT9Hs?p=preview

1. Open the datetime picker at the top of the page
2. Change the year to 2014. The valid values of the month and day picker columns don't update
3. Change the year to 2015. Now the picker columns update

**Related code:**

**Other information:**
This is being caused by the logic in component/picker/picker-column.ts:
```
if (this.lastIndex === undefined) {
  // have not set a last index yet
  this.lastIndex = this.col.selectedIndex;

} else if (this.lastIndex !== this.col.selectedIndex) {
  // new selected index has changed from the last index
  // update the lastIndex and emit that it has changed
  this.lastIndex = this.col.selectedIndex;
  var ionChange = this.ionChange;
  if (ionChange.observers.length > 0) {
    this._zone.run(ionChange.emit.bind(ionChange, this.col.options[this.col.selectedIndex]));
  }
}
```

This prevents ionChange from firing the first time a picker column changes. The DateTime subscribes to this event in order to update the valid values of the picker columns. I tried making ionChange fire in both cases, can confirm it fixes the problem. I'm using a fork with this fix in my app and am not experiencing any other problems.

It's also what's causing #12070.

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
global packages:

    @ionic/cli-utils : 1.4.0
    Ionic CLI        : 3.4.0

System:

    Node       : v8.1.2
    OS         : macOS Sierra
    Xcode      : Xcode 8.3.3 Build version 8E3004b
    ios-deploy : 1.9.1
    ios-sim    : not installed
    npm        : 5.0.3
```

**Comment by eisene** [comment]
I can make a pull request with the fix I outlined in the issue, but I'm not sure what the intended behavior of picker-column is

**Comment by eisene** [comment]
It's been more than 2 weeks. Bump

**Comment by sebrojas14** [comment]
I have the same problem. In this example, I set the min time to 2017-10-02T11:00:00
![image](https://user-images.githubusercontent.com/1771574/31110551-6f723d82-a7cd-11e7-89fc-829537a0acca.png)

I changed the day to 3, and as you can see, the hours colun didn't change:
![image](https://user-images.githubusercontent.com/1771574/31110541-69d3c3dc-a7cd-11e7-9307-bf9ed24f64fc.png)
Then, I return he day o 2, and again o 3. but this time, the hours columns changed:
![image](https://user-images.githubusercontent.com/1771574/31110611-d9bf5dfa-a7cd-11e7-8a1f-ba36419cf637.png)

**Comment by eisene** [comment]
I have a fix for this that we've been using for months now without issue. I'm happy to make a pull request. I was hoping to first get some info on the intended behavior from the core dev team but I can also just make the request.

**Comment by christianguevara** [comment]
Hey @eisene can you please share your solution ?

**Comment by eisene** [comment]
Cool. Very busy for a bit, will try to make a pull request end of next week.

**Comment by devnetics** [comment]
Hey @eisene were you planning on creating the PR? If not I'm going to send one (and I'll just credit you with the fix in the PR).

**Comment by devnetics** [comment]
I pushed up a PR and credited you there (linked above. If you find the time, feel free to open your own PR and I'll close mine, so that you get some cool points on github if it gets accepted.

**Comment by wnabil** [comment]
PR works for me also, Thanks

**Comment by alexnu** [comment]
I see that the PR above was closed. So I guess this is not going to be fixed?

**Comment by satbirjhuti** [comment]
I've created a new PR for v4 with the same fix: #14329
